blocks/basic/test/qa_Selector.cpp

```diff
@@ -115,140 +122,177 @@ const boost::ut::suite SelectorTest = [] {
             .mapping                     = {{1, 0}, {1, 1}, {1, 2}},                            //
             .inValues                    = {{1}, {2}, {3}},                                     //
             .outValues                   = {{2, 2, 2, 2, 2}, {2, 2, 2, 2, 2}, {2, 2, 2, 2, 2}}, //
-            .monitorSource               = -1U,                                                 //
-            .monitorValues               = {},                                                  //
+            .inTags                      = {{tag1}, {tag2}, {tag3}},                            //
+            .outTags                     = {{tag2}, {tag2}, {tag2}},
+            .monitorSource               = -1U, //
+            .monitorValues               = {},  //
             .backPressure                = false,
             .ignoreOrder                 = false});
     };
 
     // tests with the back pressure
 
-    "Selector<T> 1 to 1 mapping, with back pressure"_test = [] {
+    "Selector<T> 1 to 1 mapping, with back pressure"_test = [tag1, tag2, tag3] {
         execute_selector_test({.nSamples = 5,                                                   //
             .mapping                     = {{0, 0}, {1, 1}, {2, 2}},                            //
             .inValues                    = {{1}, {2}, {3}},                                     //
             .outValues                   = {{1, 1, 1, 1, 1}, {2, 2, 2, 2, 2}, {3, 3, 3, 3, 3}}, //
-            .monitorSource               = -1U,                                                 //
-            .monitorValues               = {},                                                  //
+            .inTags                      = {{tag1}, {tag2}, {tag3}},                            //
+            .outTags                     = {{tag1}, {tag2}, {tag3}},
+            .monitorSource               = -1U, //
+            .monitorValues               = {},  //
             .backPressure                = true,
             .ignoreOrder                 = false});
     };
 
-    "Selector<T> only one input used, with back pressure"_test = [] {
+    "Selector<T> only one input used, with back pressure"_test = [tag1, tag2, tag3] {
         execute_selector_test({.nSamples = 5,                         //
             .mapping                     = {{1, 1}},                  //
             .inValues                    = {{1}, {2}, {3}},           //
             .outValues                   = {{}, {2, 2, 2, 2, 2}, {}}, //
-            .monitorSource               = -1U,                       //
-            .monitorValues               = {},                        //
+            .inTags                      = {{tag1}, {tag2}, {tag3}},  //
+            .outTags                     = {{}, {tag2}, {}},
+            .monitorSource               = -1U, //
+            .monitorValues               = {},  //
             .backPressure                = true,
```

Comment:
please check remaining samples in the unconnected ports, there should be '5' in this case. Similar for the cases below.